pkg/io/video/convert.go

```diff
@@ -24,15 +24,18 @@ func ToI420(r Reader) Reader {
 				addrSrc := i * 2 * yuvImg.CStride
 				addrDst := i * yuvImg.CStride / 2
 				for j := 0; j < yuvImg.CStride/2; j++ {
-					cb := uint16(yuvImg.Cb[addrSrc+j]) + uint16(yuvImg.Cb[addrSrc+yuvImg.CStride+j]) +
-						uint16(yuvImg.Cb[addrSrc+j+1]) + uint16(yuvImg.Cb[addrSrc+yuvImg.CStride+j+1])
-					cr := uint16(yuvImg.Cr[addrSrc+j]) + uint16(yuvImg.Cr[addrSrc+yuvImg.CStride+j]) +
-						uint16(yuvImg.Cr[addrSrc+j+1]) + uint16(yuvImg.Cr[addrSrc+yuvImg.CStride+j+1])
+					j2 := j * 2
+					cb := uint16(yuvImg.Cb[addrSrc+j2]) + uint16(yuvImg.Cb[addrSrc+yuvImg.CStride+j2]) +
+						uint16(yuvImg.Cb[addrSrc+j2+1]) + uint16(yuvImg.Cb[addrSrc+yuvImg.CStride+j2+1])
+					cr := uint16(yuvImg.Cr[addrSrc+j2]) + uint16(yuvImg.Cr[addrSrc+yuvImg.CStride+j2]) +
+						uint16(yuvImg.Cr[addrSrc+j2+1]) + uint16(yuvImg.Cr[addrSrc+yuvImg.CStride+j2+1])
 					yuvImg.Cb[addrDst+j] = uint8(cb / 4)
 					yuvImg.Cr[addrDst+j] = uint8(cr / 4)
 				}
 			}
 			yuvImg.CStride = yuvImg.CStride / 2
+			yuvImg.Cb = yuvImg.Cb[:yuvImg.CStride*(h/2)]
+			yuvImg.Cr = yuvImg.Cr[:yuvImg.CStride*(h/2)]
```

Comment:
Probably also limit the cap so that they can't go beyond it,

```
yuvImg.Cb[:yuvImg.CStride*(h/2):yuvImg.CStride*(h/2)]
```

Reply:
addressed!